File: app/src/main/java/io/github/freewebmovement/igniter/common/UnifyVersions.java
package io.github.freewebmovement.igniter.common;

import android.content.Intent;
import android.os.Build;

public class UnifyVersions {

    public static <T> T getParcel(Intent intent, String key, Class<T> tClass) {
        T temp;
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            temp = intent.getParcelableExtra(key, tClass);
        } else {
            temp = intent.getParcelableExtra(key);
        }
        return temp;
    }
}

File: app/src/main/java/io/github/freewebmovement/igniter/common/UnifyVersions.kt
package io.github.freewebmovement.igniter.common

import android.content.Intent
import android.os.Build
import android.os.Parcelable

object UnifyVersions {

    fun <T : Parcelable> getParcelable(intent: Intent, key: String, tClass: Class<T>): T {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            intent.getParcelableExtra(key, tClass)!!
        } else {
            @Suppress("DEPRECATION")
            intent.getParcelableExtra<Parcelable>(key) as T
        }
    }
}
